### PR TITLE
GAWB-798: pass through 'get refresh token date' endpoint

### PR DESCRIPTION
**Pull request by MattPutnam**
Use this endpoint to see if there's a refresh token.
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] **Submitter**: Tell tech lead that the PR exists if s/he wants to look at it
- [x] **Submitter**: Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [x] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x] **Tech lead**:  sign off
- [x] **LR**: sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Squash commits, rebase if necessary
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Merge to develop 
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: Check configuration files in Jenkins in case they need changes
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Verify swagger UI on dev server still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] ~~**Submitter**: Mark JIRA issue as resolved once this checklist is completed~~  Don't do this, story isn't done yet

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/firecloud/service/UserService.scala [hunk 1]
```diff
@@ -40,6 +40,9 @@ object UserService {
   val billingPath = FireCloudConfig.Rawls.authPrefix + "/user/billing"
   val billingUrl = FireCloudConfig.Rawls.baseUrl + billingPath
 
+  val refreshTokenDatePath = FireCloudConfig.Rawls.authPrefix + "/user/refreshTokenDate"
+  val refreshTokenDateUrl = FireCloudConfig.Rawls.baseUrl + refreshTokenDatePath
+
```
Comment (davidangb) [inline src/main/scala/org/broadinstitute/dsde/firecloud/service/UserService.scala]: OAuthService already defines these (OAuthService.remoteTokenDatePath) ... can you centralize, or point one at the other?

File: src/test/scala/org/broadinstitute/dsde/firecloud/service/UserServiceSpec.scala [hunk 2]
```diff
@@ -68,6 +68,13 @@ class UserServiceSpec extends ServiceSpec with UserService {
       )
 
     workspaceServer
+      .when(request.withMethod("GET").withPath(UserService.refreshTokenDatePath))
```
Comment (rushtong) [inline src/test/scala/org/broadinstitute/dsde/firecloud/service/UserServiceSpec.scala]: This also needs to be changed to use `OAuthService.remoteTokenDateUrl`

**Comment by rushtong** [comment]
@MattPutnam - One small compilation fix and it gets my thumbs-up.

**Comment by rushtong** [comment]
👍

File: src/test/scala/org/broadinstitute/dsde/firecloud/service/UserServiceSpec.scala [hunk 3]
```diff
@@ -146,6 +153,15 @@ class UserServiceSpec extends ServiceSpec with UserService {
       }
     }
 
+    "when calling GET for user refresh token date service" - {
+      "MethodNotAllowed response is not returned" in {
+        Get("/api/profile/refreshTokenDate") ~> dummyUserIdHeaders(uniqueId) ~> sealRoute(routes) ~> check {
+          log.debug("/api/profile/refreshTokenDate: " + status)
```
Comment (davidangb) [inline src/test/scala/org/broadinstitute/dsde/firecloud/service/UserServiceSpec.scala]: probably unnecessary log statement

**Comment by davidangb** [comment]
👍 one minor comment inline to remove a log statement

File: src/test/scala/org/broadinstitute/dsde/firecloud/service/UserServiceSpec.scala (hunk 3 above)
Comment (MattPutnam) [inline src/test/scala/org/broadinstitute/dsde/firecloud/service/UserServiceSpec.scala]: I was just mirroring the other test cases.

**Pull request merged**